tests/array_split.py

```diff
@@ -22,127 +22,110 @@
-    for i in range(min_size, max_size):
-        for j in range(min_size, max_size):
-            for k in range(min_size, max_size):
-                a = np.random.randn(i, j, k)
-                for axis in range(a.ndim):
-                    input_arr = []
-                    even_div = None  # a.shape[axis]
-                    uneven_div = None  # a.shape[axis]
-                    for div in range(1, (int)(math.sqrt(a.shape[axis]) + 1)):
-                        if a.shape[axis] % div == 0:
-                            even_div = div
-                        else:
-                            uneven_div = div
-                        if even_div is not None and uneven_div is not None:
-                            break
-                    if even_div is None:
-                        even_div = a.shape[axis]
-                    # divisible integer
-                    input_arr.append(even_div)
-                    # indivisble integer
-                    if even_div != uneven_div and uneven_div is not None:
-                        input_arr.append(uneven_div)
-                    # integer larger than shape[axis]
-                    input_arr.append(a.shape[axis] + np.random.randint(1, 10))
-                    # indices array which has points
-                    # within the target dimension of the src array
-                    if a.shape[axis] > 1:
-                        input_arr.append(
-                            list(range(1, a.shape[axis], even_div))
-                        )
-                    # indices array which has points
-                    # out of the target dimension of the src array
-                    input_arr.append(
-                        list(
-                            range(
-                                0,
-                                a.shape[axis]
-                                + even_div * np.random.randint(1, 10),
-                                even_div,
-                            )
-                        )
+
+    input_shapes = [(0), (1), (1, 1), (1, 1, 1), (dim, dim), (dim, dim, dim)]
+    for shape in input_shapes:
+        a = np.random.randint(low=0, high=100, size=shape)
+        for axis in range(a.ndim):
+            input_arr = []
+            even_div = None  # a.shape[axis]
+            uneven_div = None  # a.shape[axis]
+            for div in range(1, (int)(math.sqrt(a.shape[axis]) + 1)):
+                if a.shape[axis] % div == 0:
+                    even_div = div
+                else:
+                    uneven_div = div
+                if even_div is not None and uneven_div is not None:
+                    break
+            if even_div is None:
+                even_div = a.shape[axis] if (a.shape[axis] >= 1) else 1
+            # divisible integer
+            input_arr.append(even_div)
+            # indivisble integer
+            if even_div != uneven_div and uneven_div is not None:
```

Comment:
the first condition seems very confusing. I believe `even_div` and `uneven_div` meant to be different, but why are you checking if they are different?

Reply:
It can both 1 if a.shape[axis] is prime.

Reply:
ok then is there reason that you don't test with `even_div == a.shape[axis]` in that case? I feel this is another unnecessary complication in this test that is just confusing.